### ISSUE TITLE
issue about the sucess and precision in UAV

### ISSUE THREAD

**Issue by JackjackFan** [chenxin-dlut/TransT#65]
Nice work.我尝试复现您的代码时发现，使用ddp框架，4卡2080ti训练800epoch后，在UAV上的sucess和precision分别为0.680和0.880，和您论文中的结果0.691有所差异。于是我尝试使用您提供的Transt.pth进行测试，发现您提供的Transt.pth在UAV上的两个指标分别为0.680和0.875。请问这是为什么呢？为什么和您论文中提供的精度相比要差呢？麻烦您解答了- -，非常感谢！

**Comment by chenxin-dlut** [comment]
hello，请问你是用哪个toolkit测的呢？我建议用got10k的toolkit来验证结果。pysot的好像有点bug

**Comment by JackjackFan** [comment]
> hello，请问你是用哪个toolkit测的呢？我建议用got10k的toolkit来验证结果。pysot的好像有点bug

我使用的是您readme里建议的pysot的toolkit测试的，好的，我试一下got10k的toolkit验证一下。非常感谢

**Comment by chenxin-dlut** [comment]
pysot验证uav和nfs这两个数据集都有点问题，测试是没问题的，算AUC有点问题，用pysot测的结果可以直接换got10k来验证一下

**Comment by JackjackFan** [comment]
> pysot验证uav和nfs这两个数据集都有点问题，测试是没问题的，算AUC有点问题，用pysot测的结果可以直接换got10k来验证一下

好的，非常感谢您耐心解答。

**Comment by JackjackFan** [comment]
您好，使用got10k的toolkit验证结果和您论文结果一致。再次感谢您。

**Comment by iminfine** [comment]
你好，请问你用ddp4卡训练花费多长时间？
我尝试用ddp训练，速度提高并不明显

**Comment by iminfine** [comment]
我也发现这个不一致的问题，我倾向于相信pysot计算出来的结果。我用pytracking自带的评估工具计算readme中下载的uav的raw results，success为0.677。与pysot的结果相差不大。

**Comment by chenxin-dlut** [comment]
> 你好，请问你用ddp4卡训练花费多长时间？
> 我尝试用ddp训练，速度提高并不明显

我没用ddp4卡训过，我用8卡训能提升3倍

**Comment by chenxin-dlut** [comment]
> 我也发现这个不一致的问题，我倾向于相信pysot计算出来的结果。我用pytracking自带的评估工具计算readme中下载的uav的raw results，success为0.677。与pysot的结果相差不大。

我用pysot测以前的很多tracker提供的raw results，都会比他们论文里的指标低，用got10k来测就跟他们论文中一致，所以我当时感觉可能got10k的更可信些。总之只要统一比较的标准就行。

**Comment by iminfine** [comment]
> > 你好，请问你用ddp4卡训练花费多长时间？
> > 我尝试用ddp训练，速度提高并不明显
> 
> 我没用ddp4卡训过，我用8卡训能提升3倍

我看到您在这里的回复。I use the same learning rate for 8gpu, training 500 epochs with 80000 samples each epoch, with batch size 128, and the learning rate drops at 400th epoch.
_Originally posted by @chenxin-dlut in https://github.com/chenxin-dlut/TransT/issues/43#issuecomment-850772911_
请问这里您用的是ddp训练吗？训练下来一共花费多长时间？

**Comment by chenxin-dlut** [comment]
是ddp，训练大概3天

**Comment by iminfine** [comment]
> > > 你好，请问你用ddp4卡训练花费多长时间？
> > > 我尝试用ddp训练，速度提高并不明显
> > 
> > 
> > 我没用ddp4卡训过，我用8卡训能提升3倍
> 
> 我看到您在这里的回复。I use the same learning rate for 8gpu, training 500 epochs with 80000 samples each epoch, with batch size 128, and the learning rate drops at 400th epoch.
> _Originally posted by @chenxin-dlut in [#43 (comment)](https://github.com/chenxin-dlut/TransT/issues/43#issuecomment-850772911)_
> 请问这里您用的是ddp训练吗？训练下来一共花费多长时间？

我目前也是8个卡进行ddp训练（8个V100），按照您这里的推荐参数训练，大概一个小时完成一个epoch，慢的离谱=，=。
可能是我写的ddp代码有问题。

方便的话，您能否更新下项目代码，使其支持ddp？

**Comment by chenxin-dlut** [comment]
好，我之后会更新一版

**Comment by iminfine** [comment]
谢谢~

**Comment by JackjackFan** [comment]
> 我也发现这个不一致的问题，我倾向于相信pysot计算出来的结果。我用pytracking自带的评估工具计算readme中下载的uav的raw results，success为0.677。与pysot的结果相差不大。

我用pysot测出来的结果是0.680，但是论文时0.691，所以我认为是不一致的

**Comment by iminfine** [comment]
谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack ***@***.***>写道：



你好，请问你用ddp4卡训练花费多长时间？

我尝试用ddp训练，速度提高并不明显



我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同

**Comment by JackjackFan** [comment]
> 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack ***@***.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

**Comment by iminfine** [comment]
> > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

谢谢~我训练的时候FPS不到2，很不正常。我的邮箱为bmgao@hotmail.com。

**Comment by LeungWaiHo** [comment]
> > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

请问也可以发一份ddp代码给我吗？谢谢，我的邮箱是leungwaiho0515@163.com

**Comment by kongbia** [comment]
> > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

请问也可以发一份ddp代码给我吗？谢谢，我的邮箱是 zjphust@gmail.com

**Comment by FMZCY** [comment]
> > > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> > 
> > 
> > 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常
> 
> 谢谢~我训练的时候FPS不到2，很不正常。我的邮箱为[bmgao@hotmail.com](mailto:bmgao@hotmail.com)。

请问也可以发一份ddp代码给我吗？谢谢，我的邮箱是zqygreat@163.com

**Comment by FMZCY** [comment]
> > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

请问也可以发一份ddp代码给我吗？谢谢，我的邮箱是zqygreat@163.com

**Comment by kongbia** [comment]
> 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显

我用ddp 4卡 num_work=4训练，和不用ddp 4卡 num_work=16花费的时间差不多

**Comment by ChunleiWang004** [comment]
> > 谢谢，方便的话能否分享下您的DDP代码我写的DDP不知道哪里出了问题，训练的过程慢的离谱2021年8月4日18：09，鱼杰克_**@**_。***>写道：你好，请问你用ddp4卡训练消费多长时间？ —您收到此消息是因为您发表了评论。直接回复此电子邮件、在 GitHub 上查看或取消订阅。使用 iOS 或 Android 版 GitHub Mobile 随时随地进行分类通知。
> 
> 好的，我可以保存你的数据。不过你可以先看看你训练时的fps是多少，fps正常不正常



> > 谢谢，方便的话能否分享下您的DDP代码我写的DDP不知道哪里出了问题，训练的过程慢的离谱2021年8月4日18：09，鱼杰克_**@**_。***>写道：你好，请问你用ddp4卡训练消费多长时间？ —您收到此消息是因为您发表了评论。直接回复此电子邮件、在 GitHub 上查看或取消订阅。使用 iOS 或 Android 版 GitHub Mobile 随时随地进行分类通知。
> 
> 好的，我可以保存你的数据。不过你可以先看看你训练时的fps是多少，fps正常不正常

您好，可以也给我发一份DDP的代码么？我的邮箱是 736544589@qq.com

**Comment by ChunleiWang004** [comment]
> 干得好。我尝试复现您的代码时发现，使用ddp框架，4卡2080ti训练800epoch后，batch size设置为32，numwork为6，在无人机上的成功和精度分别为0.680和0.880，和您的论文中的结果 0.691 有所不同。于是我尝试使用您提供的 Transt.pth 进行测试，发现您提供的 Transt.pth 在无人机上的指标分别为 0.680 和 0.875。请问这是为什么？为什么和您两个论文中提供的精确模拟要差呢？麻烦您解答了- -，非常感谢！

您好，可以也给我发一下DDP的代码么？我的邮箱736544589@qq.com

**Comment by DavidZhangdw** [comment]
> > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

请问也可以发一份ddp代码给我吗？谢谢，我的邮箱是 [836477361@qq.com](mailto:836477361@qq.com)

**Comment by cmj18** [comment]
> > 谢谢，方便的话能否分享下您的ddp代码？我写的ddp不知道哪里出了问题，训练的过程慢的离谱2021年8月4日 18:09，Fish Jack _**@**_.***>写道： 你好，请问你用ddp4卡训练花费多长时间？ 我尝试用ddp训练，速度提高并不明显 我ddp4卡2080ti 32batch训练5天左右，训了800/400个epoch，其他的参数和论文相同 —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> 好的，我可以邮箱发给你。不过你可以先看看你训练时的fps是多少，fps正常不正常

hello，能参考一下您的ddp训练代码么，我的邮箱是1468139558@qq.com，谢谢啦~
